microsoft/testsuites/vm_extensions/kvvm_extension.py

```diff
@@ -0,0 +1,220 @@
+        # VM attributes
+        node_context = get_node_context(node)
+
+        # User's attributes
+        user_tenant_id = os.environ.get("tenant_id")
+        if user_tenant_id is None:
+            raise ValueError("Environment variable 'tenant_id' is not set.")
+        user_object_id = os.environ.get("user_object_id")
+        if user_tenant_id is None:
+            raise ValueError("Environment variable 'user_object_id' is not set.")
+        assert user_object_id is not None
+
+        # Object ID System assignment
+        object_id_vm = add_system_assign_identity(
+            credential=platform.credential,
+            subscription_id=node_context.subscription_id,
+            resource_group_name=node_context.resource_group_name,
+            vm_name=node_context.vm_name,
+            location=node_context.location,
+            log=log,
+        )
+
+        # Create Key Vault
+        keyvault_result = create_keyvault(
+            platform.credential,
+            platform.subscription_id,
+            user_tenant_id,
+            user_object_id,
+            object_id_vm,
+            node_context.location,
+            node_context.resource_group_name,
+            vault_name,
+        )
+
+        log.info(f"Created Key Vault {keyvault_result.properties.vault_uri}")
+
+        # Certificates
+        log.info(
+            "About to call create_certificates with vault_url: "
+            f"{keyvault_result.properties.vault_uri}"
+        )
+        certificate1_secret_id = create_certificate(
+            vault_url=keyvault_result.properties.vault_uri,
+            credential=platform.credential,
+            log=log,
+            cert_name="Cert1",
+        )
+        certificate2_secret_id = create_certificate(
+            vault_url=keyvault_result.properties.vault_uri,
+            credential=platform.credential,
+            log=log,
+            cert_name="Cert2",
+        )
+        log.info(
+            f"Certificates created. Cert1 ID: {certificate1_secret_id}, "
+            f"Cert2 ID: {certificate2_secret_id}"
+        )
+        assert_that(certificate1_secret_id).described_as(
+            "First certificate created successfully"
+        ).is_not_none()
+        assert_that(certificate2_secret_id).described_as(
+            "Second certificate created successfully"
+        ).is_not_none()
+
+        # Extension
+        extension_name = "KeyVaultForLinux"
+        extension_publisher = "Microsoft.Azure.KeyVault"
+        extension_version = "2.0"
+        settings = {
+            "secretsManagementSettings": {
+                "autoUpgradeMinorVersion": True,
+                "enableAutomaticUpgrade": True,
+                "pollingIntervalInS": "360",
+                "certificateStoreLocation": "/var/lib/waagent/Microsoft.Azure.KeyVault",
+                "observedCertificates": [
+                    certificate1_secret_id,
+                    certificate2_secret_id,
+                ],
```

Comment:
certificates_secret_id: List[str]
    for cert_name in ["Cert1", "Cert2"]:
        certificate_secret_id = create_certificate(
            vault_url=keyvault_result.properties.vault_uri,
            credential=platform.credential,
            log=log,
            cert_name=cert_name,
        )
        log.info(
            f"Certificates created. Cert ID: {certificate_secret_id}, "
        )
        assert_that(certificate_secret_id).described_as(
            "First certificate created successfully"
        ).is_not_none()
        certificates_secret_id.append(certificate_secret_id)

    # Extension
    extension_name = "KeyVaultForLinux"
    extension_publisher = "Microsoft.Azure.KeyVault"
    extension_version = "2.0"
    settings = {
        "secretsManagementSettings": {
            "autoUpgradeMinorVersion": True,
            "enableAutomaticUpgrade": True,
            "pollingIntervalInS": "360",
            "certificateStoreLocation": "/var/lib/waagent/Microsoft.Azure.KeyVault",
            "observedCertificates": [
                certificates_secret_id[0],
                certificates_secret_id[1],
            ],
        }
    }